crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1862,23 +1862,24 @@ macro_rules! impl_anytuple_fetch {
             }
 
             fn update_component_access(state: &Self::State, _access: &mut FilteredAccess<ComponentId>) {
-                let ($($name,)*) = state;
+                // update the access (add the read/writes)
+                <($(Option<$name>,)*)>::update_component_access(state, _access);
 
+                // update the filters (Or<(With<$name>,)>)
+                let ($($name,)*) = state;
                 let mut _new_access = _access.clone();
                 let mut _not_first = false;
                 $(
                     if _not_first {
                         let mut intermediate = _access.clone();
                         $name::update_component_access($name, &mut intermediate);
                         _new_access.append_or(&intermediate);
-                        _new_access.extend_access(&intermediate);
                     } else {
                         $name::update_component_access($name, &mut _new_access);
```

Comment:
The panic is on this line. Probably needs to be changed to use an intermediate access? But not sure we need the _not_first check anymore.

Reply:
You're right, the panic is on the line you showed.. there must be some internal use of AnyOf in the engine that's causing this panic.
The tests are probably not strong enough, let me try to trigger it in a unit test

Reply:
My change actually doesn't work with `AnyOf<&mut A>` because it's going to add the write access for A twice...

Reply:
Is your comment above still true?

Reply:
No everything should now work!